### PR TITLE
Unit tests for AutoCompletionController

### PR DESCRIPTION
**Pull request by HuanhuanSunMSFT**
When we have nested completion supported, these unit tests would help more.
@EricMSFT @AndreSayreMSFT @Microsoft/poshtools

### CODE REVIEW

File: PowerShellTools.Test/IntelliSense/AutoCompletionControllerUnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+            autoCompletionController.SetAutoCompleteState(true);
+
+            _mockedScript = @"MyFunc()";
+            _mockedCaret = 7;
+            int actual = autoCompletionController.ProcessKeystroke((uint)VSConstants.VSStd2KCmdID.RETURN);
+            Assert.AreEqual<int>(VSConstants.S_FALSE, actual);
+        }
+
+        [TestMethod]
+        public void ShouldProcessReturnKeyWhenInPowershellCode()
```
Comment (ASayre) [inline PowerShellTools.Test/IntelliSense/AutoCompletionControllerUnitTests.cs:145]: nit: Capitalization of PowerShell

File: PowerShellTools.Test/IntelliSense/AutoCompletionControllerUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+                                        });
+
+            editorOperations.Setup(m => m.MoveToEndOfLine(false))
+                            .Callback(() =>
+                                        {
+                                            // Should move caret to the end, here we remain it still as it's already at the end.
+                                        });
+
+            // {\r\n|\r\n} --> {\r\n  |\r\n}
+            editorOperations.Setup(m => m.Indent())
```
Comment (ASayre) [inline PowerShellTools.Test/IntelliSense/AutoCompletionControllerUnitTests.cs:362]: When we have our SmartIndenter updated, will this need to be updated?

File: PowerShellTools/Intellisense/AutoCompletionController.cs [hunk 3]
```diff
@@ -177,6 +175,12 @@ private int ProcessKeystroke(uint nCmdID, IntPtr pvaIn)
             return VSConstants.S_FALSE;
         }
 
+        internal void SetAutoCompleteState(bool isAutoComplete)
```
Comment (ASayre) [inline PowerShellTools/Intellisense/AutoCompletionController.cs:178]: Comments for non-private func (What's the Autocomplete state?)

File: PowerShellTools.Test/IntelliSense/AutoCompletionControllerUnitTests.cs (hunk 2 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools.Test/IntelliSense/AutoCompletionControllerUnitTests.cs:362]: Yes. Besides this, we'll need to add some logic here when we have nested braces/quotes supported.

File: PowerShellTools/Intellisense/AutoCompletionController.cs (hunk 3 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools/Intellisense/AutoCompletionController.cs:178]: It's a state indicating last keystroke. It needs to be accessed to set so we can test different scenarios.

File: PowerShellTools.Test/IntelliSense/AutoCompletionControllerUnitTests.cs (hunk 1 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools.Test/IntelliSense/AutoCompletionControllerUnitTests.cs:145]: Fixed.

**Comment by ASayre** [comment]
After the commit that includes the above fixes you mentioned, looks good :)

**Pull request merged**